packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesDevLauncherController.kt

```diff
@@ -78,8 +78,11 @@ class UpdatesDevLauncherController(
   }
 
   @get:Synchronized
-  override val launchAssetFile: String
-    get() = throw Exception("IUpdatesController.launchAssetFile should not be called in dev client")
+  override val launchAssetFile: override val launchAssetFile: String?
+    get() {
+      logger.warn("IUpdatesController.launchAssetFile should not be called in dev client, except during Detox testing")
```

Comment:
```suggestion
      logger.warn("launchAssetFile should not be called from expo-dev-client build, except for Detox testing")
```

nit: little fine tune of the message.